alf/algorithms/trac_algorithm.py

```diff
@@ -187,14 +192,12 @@ def _update_total_dists(new_action, exp, total_dists):
             exp = tf.nest.map_structure(lambda x: x.read(t), exp_array)
             state = common.reset_state_if_necessary(
                 state, initial_state, exp.step_type == StepType.FIRST)
-            time_step = ActionTimeStep()
-            time_step = time_step._replace(
-                observation=exp.observation,
-                step_type=exp.step_type,
-            )
-            policy_step = self._ac_algorithm.predict(
+            time_step = ActionTimeStep(
+                observation=exp.observation, step_type=exp.step_type)
+            policy_step = self._ac_algorithm.rollout(
```

Comment:
Only action_distribution is needed here. rollout() has extra computations such as value.

Reply:
previous methods
1.  `_ac_algorithm_actor_network()`  assume `state` has a filed named `actor`
2. `_ac_algorithm.predict` use predict_state, but  state stored in `exp_array`

Reply:
Yes, I know this. Or do you think tensorflow can optimized away the unused computation? If so, using rollout is the cleanest way. If not, I think we still should avoid the unnecessary computation. Perhaps we can convert the train_state to predict_state before putting it to TracInfo at L110 and L117?

Reply:
converted `train_state` to `predict_state` .